### PR TITLE
Use the CMS_THREAD_SAFE macro instead of [[cms::thread_safe]]

### PR DESCRIPTION
**Pull request by fwyzard**
The Intel Compiler does not seem to support attributes with a namespace:

> warning #3924: attribute namespace "cms" is unrecognized

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
The error is unrelated to these changes.

**Comment by perrotta** [comment]
please test
(Last CMSSW_10_0_X_2017-11-28-2300 IB should be fixed now)

**Comment by Dr15Jones** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by fwyzard** [comment]
rebased

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
The suggested changes for the code checks
```
diff --git a/FWCore/MessageLogger/src/ErrorObj.cc b/FWCore/MessageLogger/src/ErrorObj.cc
index d58228c7a89..8bfb5d2e981 100644
--- a/FWCore/MessageLogger/src/ErrorObj.cc
+++ b/FWCore/MessageLogger/src/ErrorObj.cc
@@ -246,7 +246,7 @@ void ErrorObj::set( const ELseverityLevel & sev, const ELstring & id )  {
   clear();
 
   myTimestamp = time( nullptr );
-  mySerial = ++ ourSerial;
+  mySerial = ++ mySerial;
 
   setID( id );
   setSeverity( sev );
diff --git a/PhysicsTools/MVAComputer/src/AtomicId.cc b/PhysicsTools/MVAComputer/src/AtomicId.cc
index da07cd65a00..7d1bd031544 100644
--- a/PhysicsTools/MVAComputer/src/AtomicId.cc
+++ b/PhysicsTools/MVAComputer/src/AtomicId.cc
@@ -59,7 +59,7 @@ namespace PhysicsTools {
 static IdCache &getAtomicIdCache()
 {
 	CMS_THREAD_SAFE static IdCache atomicIdCache;
-	return atomicIdCache;
+	return getAtomicIdCache;
 }
 
 const char *AtomicId::lookup(const char *string) throw()
```
look wrong !

**Comment by davidlange6** [comment]
indeed - lets see if its reproducible

**Comment by davidlange6** [comment]
code-checks

**Comment by fwyzard** [comment]
Seems it is reproducibly wrong.

**Comment by davidlange6** [comment]
probably its because its using a broken IB

**Comment by slava77** [comment]
code-checks

**Comment by fwyzard** [comment]
Can you please stop asking for useless code checks, and just merge this, please ?

**Comment by slava77** [comment]
On 11/30/17 10:19 AM, Andrea Bocci wrote:
> Can you please stop asking for useless code checks, and just merge this,
> please ?
> 

current strategy is that the code-checks is blocking.
If this is merged, the next time these files are modified with
functional changes
the code-checks will fail.
IMHO, the code-checker should be fixed before this can be merged.

**Comment by fwyzard** [comment]
@slava77 I understand what you say, but I disagree with your conclusion: the lines affected by the code checks are already there before this PR, so any changes affecting those files will run into the same problem, irrespective of this PR being merged or not.

My point is that this PR does not create new code check issues, and is only a trivial clean up, so I'd rather have it merged sooner than later, and especially before any other PR that may require it to be rebased again.

**Comment by slava77** [comment]
On 11/30/17 1:46 PM, Andrea Bocci wrote:
> @slava77 <https://github.com/slava77> I understand what you say, but I
> disagree with your conclusion: the lines affected by the code checks are
> already there before this PR, so any changes affecting those files will
> run into the same problem, irrespective of this PR being merged or not.
> 
> My point is that this PR does not create new code check issues, and is
> only a trivial clean up, so I'd rather have it merged sooner than later,
> and especially before any other PR that may require it to be rebased again.

in both cases the code-checks patch points to a variable which
declaration has "CMS_THREAD_SAFE" prepended, each from this PR edits.
So, it seemed to me that there is a connection

Are you saying that the code-checks fails and suggests the same nonsense
changes  without your changes?

**Comment by fwyzard** [comment]
I couldn't say.
Running `scram build code-checks` with this branch on top of CMSSW_10_0_X_2017-11-30-1100 does not spot any code problems anyway, so I cannot check if it would find any problems without the changes...

**Comment by fwyzard** [comment]
@cmsbuild, please check ...

**Comment by davidlange6** [comment]
code-checks

**Comment by davidlange6** [comment]
so indeed the problem was the underlying IB not the code checker..

File: Calibration/Tools/interface/EcalRingCalibrationTools.h [hunk 1]
```diff
@@ -48,7 +49,7 @@ class EcalRingCalibrationTools
   
   static std::atomic<bool> isInitializedFromGeometry_;
 
-  [[cms::thread_guard("isInitializedFromGeometry_")]]
+  CMS_THREAD_GUARD("isInitializedFromGeometry_")
```
Comment (ghellwig) [inline Calibration/Tools/interface/EcalRingCalibrationTools.h]: @fwyzard maybe, I am not experienced enough with macros, but given the definition of `CMS_THREAD_GUARD`, I would rather translate it to this statement:
```
CMS_THREAD_GUARD(isInitializedFromGeometry_)
```

Comment (fwyzard) [reply Calibration/Tools/interface/EcalRingCalibrationTools.h]: Good point... we are already using both syntaxes e.g. in [[1]] and [[2]].
Let me figure out which one is correct and fix the others...

[1]: https://github.com/cms-sw/cmssw/blob/master/DataFormats/Common/interface/DetSetVectorNew.h#L111
[2]: https://github.com/cms-sw/cmssw/blob/master/DataFormats/PatCandidates/interface/PackedCandidate.h#L727

Comment (fwyzard) [reply Calibration/Tools/interface/EcalRingCalibrationTools.h]: I think the correct syntax should be:
```c++
#define CMS_THREAD_GUARD(_var_) [[cms::thread_guard(#_var_)]]
...
CMS_THREAD_GUARD(isInitializedFromGeometry_)
static short endcapRingIndex_[EEDetId::IX_MAX][EEDetId::IY_MAX];
```

Comment (fwyzard) [reply Calibration/Tools/interface/EcalRingCalibrationTools.h]: The present definition
```c++
#define CMS_THREAD_GUARD(_var_) [[cms::thread_guard("#_var_")]]
```
expands to
```
[[cms::thread_guard("#_var_")]]
```
no matter the argument of the macro - which I doubt was the intended behaviour.

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by ghellwig** [comment]
+1
provided jenkins agrees

**Comment by Dr15Jones** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/21480	69ed6b8
- technical update
- jenkins tests pass

**Comment by davidlange6** [comment]
merge

**Pull request merged**